### PR TITLE
device: Add Nextthing CHIP pro

### PR DESCRIPTION
**Pull request by mripard**
The kernel with LKDTM doesn't seem to boot for some reason, otherwise it
all works.

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>

### CODE REVIEW

**Comment by mripard** [comment]
This was tested on our test lab with all 4.13 kernels configurations available

**Comment by khilman** [comment]
What about older stable kernels?  We like to avoid new boards showing up as new failures in stable when they are known not to work, so please blacklist as necessary.  Or, if the DT isn't present in any stable trees, then it won't get boot tested and should be fine.  Please let us know.

**Comment by khilman** [comment]
New kernel CI platforms need upstream LAVA support.  Please submit the device-type support to upstream lava-server, or clarify the status of upstream LAVA support for this board.  Thanks.

**Comment by mripard** [comment]
It should work in all stable releases since 4.10, release at which the DT was introduced. It's always been known to work since, and any error needs to be addressed anyway.

I'll submit the device-type tomorrow.

**Comment by mripard** [comment]
I just sent a pull request to lava-server:
https://github.com/Linaro/lava-server/pull/10

**Comment by mripard** [comment]
And it's been merged now
https://git.linaro.org/lava%2Flava-server.git/commit/?id=80d1a9fa05d91c031b1a78555880a9ae26b23e5e

File: lib/device_map.py [hunk 1]
```diff
@@ -1422,6 +1422,17 @@
                     'fastboot': False,
                     'mach': 'mvebu'}
 
+chip_pro = {'device_type': 'sun5i-gr8-chip-pro',
+                    'templates': ['generic-uboot-tftp-ramdisk-template.jinja2'],
+                    'kernel_defconfig_blacklist': [],
+                    'defconfig_blacklist': ['allmodconfig',
+                                            'multi_v7_defconfig+CONFIG_LKDTM=y'],
+                    'kernel_blacklist': [],
```
Comment (mattface) [inline lib/device_map.py:1430]: Kevins point was that you need to blacklist older kernels, as there's no point trying to boot it on anything older than 4.10.
Please have a look at all the kernels we boot here: https://kernelci.org/job/ and blacklist any that are known to not work, thanks

**Comment by khilman** [comment]
@mattface: unless something changed recently in lava-ci,  there's no need to blacklist kernels where the DT is not present since kCI will not generate a boot job when there is no DT.  
So in this case, since the DT was introduced in v4.10, and is expected to boot from then on, I don't think anything else is needed in the blacklist.

**Comment by mattface** [comment]
Ah yeah of course, clearly I need more sleep, ok then it's ready to merge

**Pull request merged**